sphinx/ext/autodoc/__init__.py

```diff
@@ -400,7 +400,7 @@ def parse_name(self) -> bool:
         self.args = args
         self.retann = retann
         self.fullname = (self.modname or '') + \
-                        ('.' + '.'.join(self.objpath) if self.objpath else '')
+                        ('::' + '.'.join(self.objpath) if self.objpath else '')
```

Comment:
Is this change needed? I can't imagine the side effect of this change. I'm afraid that this change causes errors in 3rd party extensions.

Reply:
Removed it.  There is indeed no formal notion of "fully qualified object name" in python.